pkg/platform/platform.go

```diff
@@ -74,6 +74,13 @@ type Platform interface {
 	Meta() Meta
 }
 
+// PostApplyHook is an interface, which platforms may optionally implement. If they do so,
+// after Terraform finishes applying, kubeconfig content will be passed to this function and
+// executed. This allows running sanity checks on the newly created cluster.
```

Comment:
I'd simplify this a bit.

```suggestion
// PostApplyHook runs code after Terraform finishes applying. This allows
// running sanity checks on the newly created cluster. Implementing this
// interface is optional for platforms.
```